path.go

```diff
@@ -528,8 +528,22 @@ func (n *selectorNode) filter(node ast.Node) (ast.Node, error) {
 			}
 			return filtered, nil
 		}
+	case ast.AnchorType:
+		for _, value := range node.(*ast.AnchorNode).Value.(*ast.MappingNode).Values {
+			key := value.Key.GetToken().Value
+			if key == n.selector {
```

Comment:
Use guard to avoid deep indentation.

```
if key != n.selector {
  continue
}
```

Reply:
Fixed this by calling the filter function with node.(*ast.AnchorNode).Value